### DISCUSSION TITLE
Why is C++ symbolic optimization (NLP/IPOPT) slower than the Python version

### DISCUSSION THREAD

**Discussion by dhathris** [casadi/casadi#3913] [Q&A]
I have a general question about implementing my NLP (IPOPT) problem in Python vs C++. In my practical experience, the C++ optimizer is several orders of magnitude slower.than the Python version. What could be the reasons for it? Is it a general issue is CasADi that C++ versions tend to be slower than the Python versions or could my C++ implementation the culprit here?

**Comment by jaeandersson** [comment]
It's your implementation. Or maybe how the C++ code is compiled. Python interface uses C++ underneath.
